crates/wit-parser/src/resolve.rs

```diff
@@ -1873,17 +1928,27 @@ impl Remap {
         if let Some(spans) = spans {
             assert_eq!(iface.functions.len(), spans.funcs.len());
         }
-        for (i, (_, func)) in iface.functions.iter_mut().enumerate() {
-            if !resolve.include_stability(&func.stability) {
+        for (i, (func_name, func)) in iface.functions.iter_mut().enumerate() {
+            if !resolve
+                .include_stability(&func.stability, iface_pkg_id)
+                .with_context(|| {
+                    format!(
+                        "failed to process feature gate for function [{func_name}] in package [{}]",
+                        resolve.packages[*iface_pkg_id].name,
+                    )
+                })?
+            {
                 continue;
             }
             let span = spans.map(|s| s.funcs[i]);
             self.update_function(resolve, func, span)
                 .with_context(|| format!("failed to update function `{}`", func.name))?;
         }
-        iface
-            .functions
-            .retain(|_, f| resolve.include_stability(&f.stability));
+        iface.functions.retain(|_, f| {
+            resolve
+                .include_stability(&f.stability, iface_pkg_id)
+                .is_ok_and(|v| v)
+        });
```

Comment:
Can this be rewritten to not swallow the error? Otherwise it looks like this is not reporting any errors from `include_stability`

Reply:
Did it -- it did get pretty wild though, I'm open to any simpler refactors if you have a suggestion:

https://github.com/bytecodealliance/wasm-tools/pull/1689/commits/667de50bb72a576adf79dabec25a56fd9a398772